Use external URL for last email check

Cruikshanks opened the pull request:
https://github.com/DEFRA/water-abstraction-ui/pull/2267

We found that the endpoint `/notifications/last` was not available in our non-prod environments because of a dependence on the old `TEST_MODE` env var. When updating that, we realised we were maintaining the same logic in both the internal and external apps. We only need it in one for our acceptance tests to work, so we've deleted the external endpoint and updated the internal one.

That means we need to update the custom command that uses it.

timstone commented on features/page_objects/licences_page.rb:
```diff
@@ -3,11 +3,31 @@ class LicencesPage < SitePrism::Page
 
   element(:changepw, ".header-links a:nth-child(1)")
-  elements(:licences, ".license-result:nth-child(128) .license-result__column")
+  # elements(:licences, ".license-result:nth-child(128) .license-result__column")
```
Removed commented out lines unless they are for information

timstone commented on features/step_definitions/search_filter_steps.rb:
```diff
@@ -0,0 +1,84 @@
+
+Given(/^all licences containing that term are shown on screen$/) do
+  @licences_page = @front_app.licences_page
+  puts "Licences count is:   " + @front_app.licences_page.view_links.count.to_s
```
Again remove the puts

timstone commented on features/step_definitions/search_filter_steps.rb:
```diff
@@ -0,0 +1,84 @@
+end
+
+Given(/^I can see the original number of licences$/) do
+  puts "Licences count is: " + @front_app.licences_page.view_links.count.to_s
```
Puts puts puts puts

timstone commented on features/step_definitions/search_filter_steps.rb:
```diff
@@ -0,0 +1,84 @@
+  @front_app.licences_page.serialno_link.click
+end
+
+Given(/^the table is sorted by licence number in ascending order$/) do
```
You could probably do something with having an array of elements sorting them in order and then asserting the first shown on the page is the same as the first element in the array. What you've done is ok but is hard coded to the test data so more brittle. I'm sure Alan can advise if you wan't to look into options.

andrewhick replied:
Will try and work this out for the next iteration.

Cruikshanks committed:
Use external URL for last email check

https://github.com/DEFRA/water-abstraction-ui/pull/2267

We found that the endpoint `/notifications/last` was not available in our non-prod environments because of a dependence on the old `TEST_MODE` env var. When updating that, we realised we were maintaining the same logic in both the internal and external apps. We only need it in one for our acceptance tests to work, so we've deleted the external endpoint and updated internal one.

That means we need to update our the custom command that uses it.

Cruikshanks committed:
Update request to endpoint

The internal URL is our `baseUrl` as it's what most of the tests are written against. That means we can skip needing to specify the root domain completely.

Cruikshanks committed:
Merge branch 'main' into use-external-endpoint-for-last-email-2
